File: src/main/java/cookiedragon/luchadora/module/impl/player/ReachModule.java
package cookiedragon.luchadora.module.impl.player;

import cookiedragon.luchadora.event.api.Subscriber;
import cookiedragon.luchadora.event.entity.EntityReachEvent;
import cookiedragon.luchadora.module.AbstractModule;
import cookiedragon.luchadora.module.Category;
import cookiedragon.luchadora.value.values.NumberValue;

/**
 * @author cookiedragon234 01/Jan/2020
 */
@AbstractModule.Declaration(name = "Reach", description = "Reach further", category = Category.PLAYER)
public class ReachModule extends AbstractModule
{
	private final NumberValue reachDistance = new NumberValue("Distance", 3.5f, 0, 6);
	
	@Subscriber
	private void onReach(EntityReachEvent event)
	{
		event.reachDistance = reachDistance.getValue().floatValue();
	}
	
	@Override
	protected void onEnabled()
	{
	
	}
	
	@Override
	protected void onDisabled()
	{
	
	}
}

File: src/main/kotlin/cookiedragon/luchadora/module/impl/player/ReachModule.kt
package cookiedragon.luchadora.module.impl.player

import cookiedragon.eventsystem.Subscriber
import cookiedragon.luchadora.event.entity.EntityReachEvent
import cookiedragon.luchadora.module.AbstractModule
import cookiedragon.luchadora.module.Category
import cookiedragon.valuesystem.NumberValue

/**
 * @author cookiedragon234 11/Jan/2020
 */
class ReachModule : AbstractModule("Reach", "Reach further", Category.PLAYER) {
	private val reachDistance = NumberValue("Distance", 3.5f, 0, 6)
	
	@Subscriber
	private fun onReach(event: EntityReachEvent) {
		event.reachDistance = reachDistance.value.toFloat()
	}
}
